### PR TITLE
Member

### PR DESCRIPTION
**Pull request by rootEnginear**
Figma: https://www.figma.com/file/cdtZnPegHm5CGHWqIND5Im/Design_Phase01

### CODE REVIEW

File: src/components/Assemblies/Members/Header.svelte
Comment (Th1nkK1D) [inline src/components/Assemblies/Members/Header.svelte:19]: Use `toLocaleDateString` or `dayjs` instead of manually + 543 to avoid the magic number.

File: src/components/NavigationBar/NavigationBar.svelte
Comment (Th1nkK1D) [inline src/components/NavigationBar/NavigationBar.svelte:77]: Not a clean way to do it but let's keep it for now. After the merge, I'll create a new issue to refactor this (thinking about Svelte's store) but not a priority.

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:21]: Usually `e` is for the event, isn't it? I think you can just deconstructed `({isActive}) => isActive`

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:55]: It took me a while haha. Let's use `currentCategory`, it's not that long.

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:72]: Better avoid global selection. Maybe `bind:this` to the closest parent that contains all the needed elements and query selector inside it.

```ts
step: someParentElement.querySelectorAll('.member-subcategory')
```

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:95]: Is this for not having a too-long line? No need to change, I've just never seen this before lol.

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:119]: We have the `GroupByOption` enum in `src/routes/assemblies/[id]/members/[groupby]/groupby-options.ts` which also defines the possible path. You better use this to avoid typos, or when it changes. 

PS. found several places that you can change.

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:154]: replaceAll is easier to understand and safely to use now https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

But can you concat before replace? (Or there is some case I didn't think about.)

```js
`${group.name}-${name}`.replaceAll(' ', '-')
```

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:191]: Let's use the same camel case for the variable.

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [reply src/routes/assemblies/[id]/members/[groupby]/+page.svelte:154]: And I just found several usages below, better refactor to a function like

```js
const getSubgroupHeadingId = (group, name) => `${group.name}-${name}`.replaceAll(' ', '-')
```

File: src/routes/assemblies/[id]/members/[groupby]/+page.svelte
Comment (Th1nkK1D) [inline src/routes/assemblies/[id]/members/[groupby]/+page.svelte:204]: Is it possible to just 

```svelte
{#each members as member, idx (member.id + idx)}
		<PoliticianProfile {...member} />
{/each}
```

**Review by Th1nkK1D — CHANGES_REQUESTED** [review]

File: src/components/NavigationBar/NavigationBar.svelte
Comment (rootEnginear) [reply src/components/NavigationBar/NavigationBar.svelte:77]: yea, i think it's is better to reference with store in conjunction tailwinds (`class:xxx={$navbarVisibleStore}`)

**Review by Th1nkK1D — APPROVED** [review]
LGTM! I'll merge and if designer request any changes, you can fix in the main branch.

**Comment by Th1nkK1D** [comment]
Don't forget to add the ticket name in the commit message like `feat(wev-60):  add assembly members page` so we can tract it. (This time I already squashed your commits)

**Pull request merged**
